Library/Homebrew/cmd/install.rb

```diff
@@ -396,6 +396,20 @@ def install_formula(f, args:)
     f.print_tap_action
     build_options = f.build
 
+    if f.outdated? && !f.head?
```

Comment:
```suggestion
    if Homebrew::EnvConfig.no_install_upgrade? && f.outdated? && !f.head?
```

just to avoid running this logic unnecessarily?

Reply:
Yep, but `Homebrew::EnvConfig.no_install_upgrade?` should be negated. Forgot to add this. Thanks for pointing out.